src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -535,7 +535,8 @@ private void RaiseProjectDiagnosticsCreated(Project project, StateSet stateSet,
 
             foreach (var documentId in newAnalysisResult.DocumentIds)
             {
-                var document = project.GetTextDocument(documentId);
+                var document = project.GetTextDocument(documentId)
+                    ?? project.TryGetSourceGeneratedDocumentForAlreadyGeneratedId(documentId);
```

Comment:
Why is TryGet... safe to call here? I'd strongly prefer we make the parent method async if possible and call the regular methods instead. But this otherwise needs a comment since there's some assumptions here.

Reply:
⏱️ Adding a comment

Reply:
➡️ Comment now added